nav2_collision_monitor/src/collision_monitor_node.cpp

```diff
@@ -0,0 +1,535 @@
+{
+  if (!polygon_topic_.length()) {
+    // Topic is not specified: there is no need to publish
+    return;
+  }
+
+  // Fill PolygonStamped struct
+  std::unique_ptr<geometry_msgs::msg::PolygonStamped> poly_s =
+    std::make_unique<geometry_msgs::msg::PolygonStamped>();
+  poly_s->header.stamp = this->now();
+  poly_s->header.frame_id = base_frame_id_;
+  poly_s->polygon = polygons_pub_;
+
+  // Publish polygon
+  polygon_pub_->publish(std::move(poly_s));
+}
+
+void CollisionMonitorNode::odomCallback(nav_msgs::msg::Odometry::ConstSharedPtr msg)
+{
+  std::lock_guard<mutex_t> lock(velocity_mutex_);
+
+  double dt;
+  try {
+    dt = (this->now() - msg->header.stamp).seconds();
+  } catch (const std::exception & ex) {
+    RCLCPP_WARN(
+      get_logger(),
+      "Can not get difference between collision monitor and odom time: %s",
+      ex.what());
+    dt = 0.0;
+  }
+
+  if (std::fabs(dt) <= max_time_shift_) {
+    velocity_.x = msg->twist.twist.linear.x;
+    velocity_.y = msg->twist.twist.linear.y;
+    velocity_.tw = msg->twist.twist.angular.z;
+    velocity_valid_ = true;
+  } else {
+    // Warn and ignore the odom data if it and Collision Monitor node have different time stamps
+    RCLCPP_WARN(
+      get_logger(),
+      "Odom and collision monitor node timestamps differ on %f seconds. Ignoring odom.",
+      dt);
+    velocity_valid_ = false;
+  }
+}
+
+Velocity CollisionMonitorNode::getVelocity()
+{
+  std::lock_guard<mutex_t> lock(velocity_mutex_);
+  return velocity_;
+}
+
+bool CollisionMonitorNode::velocityValid()
+{
+  std::lock_guard<mutex_t> lock(velocity_mutex_);
+  return velocity_valid_;
+}
+
+void CollisionMonitorNode::resetVelocity()
+{
+  std::lock_guard<mutex_t> lock(velocity_mutex_);
+  velocity_ = {0.0, 0.0, 0.0};
+  velocity_valid_ = false;
+}
+
+void CollisionMonitorNode::cmdVelInCallback(geometry_msgs::msg::Twist::ConstSharedPtr msg)
+{
+  std::lock_guard<mutex_t> lock(cmd_vel_in_mutex_);
+  cmd_vel_in_ = {msg->linear.x, msg->linear.y, msg->angular.z};
+
+  workerMain();
+}
+
+Velocity CollisionMonitorNode::getCmdVelIn()
+{
+  std::lock_guard<mutex_t> lock(cmd_vel_in_mutex_);
+  return cmd_vel_in_;
+}
+
+void CollisionMonitorNode::publishVelocity(const Velocity & vel)
+{
+  std::unique_ptr<geometry_msgs::msg::Twist> cmd_vel_out_msg =
+    std::make_unique<geometry_msgs::msg::Twist>();
+  cmd_vel_out_msg->linear.x = vel.x;
+  cmd_vel_out_msg->linear.y = vel.y;
+  cmd_vel_out_msg->linear.z = 0.0;
+  cmd_vel_out_msg->angular.x = 0.0;
+  cmd_vel_out_msg->angular.y = 0.0;
+  cmd_vel_out_msg->angular.z = vel.tw;
+
+  cmd_vel_out_pub_->publish(std::move(cmd_vel_out_msg));
+}
+
+bool CollisionMonitorNode::getParameters()
+{
+  try {
+    auto node = shared_from_this();
+
+    // Initializing polygon topic with empty string.
+    // If it won't be re-writed, polygons won't be published.
+    nav2_util::declare_parameter_if_not_declared(
+      node, "polygon_topic", rclcpp::ParameterValue(""));
```

Comment:
Default to something meaningful

Reply:
Done. Now it will be set to a polygon name by-default: https://github.com/ros-planning/navigation2/pull/2982/files#diff-6ff706b80d78c34ef10bd9efd8109834fe66be34241d395835af7c3857df760eR206-R207